### PR TITLE
Add Eval UI and support for embedding Ivy to EdLessons

### PR DESCRIPTION
**Pull request by cs1090218**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -210,15 +265,341 @@ def update_skill(skill_name):
 # Launch the Application
 ivy_main_page.queue()
 
-with gr.Blocks() as evaluation_page:
-    # Develop Evaluation page
-    # Placeholder interface
-    gr.Interface(lambda x: f"Hello {x}!", inputs="text", outputs="text")
+evaluation_css = open("css/evaluation.css", "r").read()
```
Comment (gracebrazil28) [inline main.py:268]: might be useful to include a check or fallback mechanism in case the CSS file fails to load

File: main.py [hunk 2]
```diff
@@ -210,15 +265,341 @@ def update_skill(skill_name):
...
+        global EVALUATION_QUESTIONS
+        global EVALUATION_QUESTION_NUM
+        eval_response_data = {
+            "Timestamp": timestamp,
+            "Username": UserConfig.USERNAME,
+            "SessionId": UserConfig.ACCESS_TOKEN,
+            "Skill": MCM_SKILL,
+            "Question": EVALUATION_QUESTIONS[EVALUATION_QUESTION_NUM][1],
+            "QuestionType": EVALUATION_QUESTIONS[EVALUATION_QUESTION_NUM][0],
+            "Response": response_text,
```
Comment (gracebrazil28) [inline main.py:496]: How is the response_text validated before it’s sent to the backend? Do we accept all inputs for now?

File: main.py [hunk 3]
```diff
@@ -210,15 +265,341 @@ def update_skill(skill_name):
...
+            "Metric_Correctness": eval_ratings[0],
+            "Metric_Completeness": eval_ratings[1],
+            "Metric_Confidence": eval_ratings[2],
+            "Metric_Comprehensibility": eval_ratings[3],
+            "Metric_Compactness": eval_ratings[4],
+        }
+        evaluation_responses_table.put_item(Item=eval_response_data)
+
+    def submit_rating_clear_update_question(response_concatenated_eval_ratings):
```
Comment (gracebrazil28) [inline main.py:505]: Do we have error handing for cases where ratings or responses might be missing or invalid

File: main.py [hunk 4]
```diff
@@ -23,15 +23,20 @@
...
+    GET_USER_INFO_URL,
+    EVALUATION_METRIC_DESCRIPTION,
 )
 
 from user_data import UserConfig
 
+MCM_SKILL = "Classification"
 MCM_URL = "https://classification.dilab-ivy.com/ivy/ask_question"
+EVALUATION_QUESTIONS = []
+EVALUATION_QUESTION_NUM = 0
```
Comment (gracebrazil28) [inline main.py:35]: Should we try to put these in constants.py?
VALUATION_QUESTIONS = []
EVALUATION_QUESTION_NUM = 0

**Review by gracebrazil28 — CHANGES_REQUESTED** [review]
Added some comments and question, looks good so far. Maybe just clean up on constants but looks ok to me

File: main.py (hunk 2 above)
Comment (cs1090218) [reply main.py:496]: The UI doesn't allow modification to the response text from the user. The response_text would be exactly what MCM/MAGE returned.

File: main.py (hunk 3 above)
Comment (cs1090218) [reply main.py:505]: I'll cover some of the edge cases in a future PR. Specifically the following cases come to mind:
1- Not all metrics were evaluated before pressing next question button. I think we should allow this behavior (thinking that user didn't want to forcibly give a response they didn't feel comfortable judging on a particular metric)
2- Question's response wasn't generated. We shouldn't allow this and I'll get to this in a future PR.

File: main.py (hunk 4 above)
Comment (cs1090218) [reply main.py:35]: Actually these will change whenever a skill is selected from the drop down menu or a response is submitted. So I didn't put them in constants.
Ideally I want to explore having these as class variables (as opposed to global variables that they are right now). I have this on my TODO.

File: main.py (hunk 1 above)
Comment (cs1090218) [reply main.py]: Good point, done.

**Review by gracebrazil28 — APPROVED** [review]
LGTM, let's roll this out for the sprint week. We'll maybe create some tickets to track our TODOs.

**Pull request merged**